plugin/pkg/scheduler/core/generic_scheduler.go

```diff
@@ -423,11 +487,322 @@ func EqualPriorityMap(_ *v1.Pod, _ interface{}, nodeInfo *schedulercache.NodeInf
+				// picked over a node with a smaller number of pods with the same negative
+				// priority (and similar scenarios).
+				sumPriorities += int64(util.GetPodPriority(pod)) + int64(math.MaxInt32+1)
+			}
+			if sumPriorities < minSumPriorities {
+				minSumPriorities = sumPriorities
+			}
+			nodeScore.sumPriorities = sumPriorities
+		}
+	}
+	if len(lowestHighPriorityNodes) == 1 {
+		return lowestHighPriorityNodes[0].node
+	}
+	// There are multiple nodes with the same minimum highest priority victim.
+	// Choose the one(s) with lowest sum of priorities.
+	minNumPods := math.MaxInt32
+	lowestSumPriorityNodes := []*nodeScore{}
+	for _, nodeScore := range lowestHighPriorityNodes {
+		if nodeScore.sumPriorities == minSumPriorities {
+			lowestSumPriorityNodes = append(lowestSumPriorityNodes, nodeScore)
+			if nodeScore.numPods < minNumPods {
+				minNumPods = nodeScore.numPods
+			}
+		}
+	}
+	if len(lowestSumPriorityNodes) == 1 {
+		return lowestSumPriorityNodes[0].node
+	}
+	// There are still more than one node with minimum highest priority victim and
+	// lowest sum of victim priorities. Find the anyone with minimum number of victims.
+	for _, nodeScore := range lowestSumPriorityNodes {
+		if nodeScore.numPods == minNumPods {
+			return nodeScore.node
+		}
+	}
+	glog.Errorf("We should never reach here!")
+	return nil
+}
+
+// selectNodesForPreemption finds all the nodes with possible victims for
+// preemption in parallel.
+func selectNodesForPreemption(pod *v1.Pod,
+	nodeNameToInfo map[string]*schedulercache.NodeInfo,
+	potentialNodes []*v1.Node,
+	predicates map[string]algorithm.FitPredicate,
+	metadataProducer algorithm.PredicateMetadataProducer,
+) (map[*v1.Node][]*v1.Pod, error) {
+
+	nodeNameToPods := map[*v1.Node][]*v1.Pod{}
+	var resultLock sync.Mutex
+
+	// We can use the same metadata producer for all nodes.
+	meta := metadataProducer(pod, nodeNameToInfo)
+	checkNode := func(i int) {
+		nodeName := potentialNodes[i].Name
+		var metaCopy algorithm.PredicateMetadata
+		if meta != nil {
+			metaCopy = meta.ShallowCopy()
+		}
+		pods, fits := selectVictimsOnNode(pod, metaCopy, nodeNameToInfo[nodeName], predicates)
+		if fits {
+			resultLock.Lock()
+			nodeNameToPods[potentialNodes[i]] = pods
+			resultLock.Unlock()
+		}
+	}
+	workqueue.Parallelize(16, len(potentialNodes), checkNode)
+	return nodeNameToPods, nil
+}
+
+func nodePassesExtendersForPreemption(
+	pod *v1.Pod,
+	nodeName string,
+	victims []*v1.Pod,
+	nodeNameToInfo map[string]*schedulercache.NodeInfo,
+	allNodes []*v1.Node,
+	extenders []algorithm.SchedulerExtender) (bool, error) {
+	// If there are any extenders, run them and filter the list of candidate nodes.
+	if len(extenders) == 0 {
+		return true, nil
+	}
+	// Remove the victims from the corresponding nodeInfo and send nodes to the
+	// extenders for filtering.
+	originalNodeInfo := nodeNameToInfo[nodeName]
+	defer func() { nodeNameToInfo[nodeName] = originalNodeInfo }()
```

Comment:
It doesn't really matter, but for me it's just easier to read whn I know what the defer is for. So I would put it right after:
```
nodeNameToInfo[nodeName] = nodeInfoCopy
```
[as long as we don't do this, this defer is not needed].

Reply:
Done.